### ISSUE TITLE
Slowness

### ISSUE THREAD

**Issue by rtrainer** [MarimerLLC/cslaforum#504]
I have an application that is running very slow. I have narrowed it down to a piece of code that retrieves some raw data from the database and then hydrates the objects:

    foreach (Bd.Icm.DataAccess.Database.Part data in parts1)
    {
        var p = Part.GetPart(data, version, true);
        parts2.Add(p);
    }

 Here is GetPart:
    public static Part GetPart(Bd.Icm.DataAccess.Database.Part data, int version, bool topLevel)
    {
        Part._topLevel = topLevel;
        return DataPortal.FetchChild<Part>(data, version);
    }

Fetch:
      ImportData(data, version);
      BusinessRules.CheckRules();

Import just loads the raw data into the object.

This is not my code. It was written by someone else a couple of years ago. My role in this is to try and speed it up.

**Comment by jonnybee** [comment]
It is impossible from what you show here to tell anything about where time is spent. 

You should profile your code to determine where time is spent and then take actions.

**Comment by rtrainer** [comment]
Yes, I realize that there isn't mush to go on with regards to my code. I have already profiled the code and the bottleneck is in the loop that contains GetPart. I inherited the code so I don't know the whys about the the way things were implemented.

So I guess what I should ask is, why types of things could be causing slowness when the objected is hydrated. It looks like there is underlying support code that is doing a bunch of LoadProperty calls but there doesn't appear to be anything that should be taking a long time.

**Comment by jonnybee** [comment]
Could be 

- ListChangedEvents (f.ex: should suppress list changed events in parts2 list)
- BusinessRules (with database lookup or long running)
- Other database access / webservice calls 

This should show up in profiling of code.

**Comment by rockfordlhotka** [comment]
Also, you don't have subqueries or entity relationships that might trigger database queries on a per-child basis do you? In other words, you do _one_ query in the parent that retrieves _all_ the data, and then just populate the child objects with that data right?
